### PR TITLE
APR-2033 : Application Withdrawal

### PR DESCRIPTION
**Pull request by martin-lush**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/GatewayReviewController.cs [hunk 1]
```diff
@@ -51,11 +52,24 @@ public async Task EvaluateGateway(Guid applicationId, [FromBody] EvaluateGateway
         {
             await _mediator.Send(new EvaluateGatewayRequest(applicationId, request.IsGatewayApproved, request.EvaluatedBy));
         }
+
+        [HttpPost("GatewayReview/{applicationId}/Withdraw")]
+        public async Task<bool> WithdrawApplication(Guid applicationId, [FromBody] GatewayWithdrawApplicationRequest request)
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/GatewayReviewController.cs:57]: Does this need to be a FromBody? Doesn't look like it.  Have a look and leave as is, if it does.

File: src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs [hunk 2]
```diff
@@ -105,7 +105,9 @@ public async Task<IActionResult> Applications()
 
             var signinId = await _userService.GetSignInId();
             var applications = await _apiClient.GetApplications(signinId, false);
-            applications = applications.Where(app => app.ApplicationStatus != ApplicationStatus.Rejected && app.ApplicationStatus != ApplicationStatus.Cancelled).ToList();
+
+            var statusFilter = new[] { ApplicationStatus.Rejected, ApplicationStatus.Cancelled, ApplicationStatus.Withdrawn };
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs:109]: leave for now, but if we're reusing this, or it's possibly going to be more complex, we need a static service that returns this list

**Review by MarkFCain — DISMISSED** [review]
no real changes, approved

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/GatewayReviewController.cs (hunk 1 above)
Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/GatewayReviewController.cs:57]: it contains comments & user info which would be too long via query string

File: src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs (hunk 2 above)
Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs:109]: agreed. This is going to get expanded on soon

**Review by MarkFCain — DISMISSED** [review]

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**